composer/callbacks/image_visualizer.py

```diff
@@ -0,0 +1,189 @@
+    +---------------------------------------------+---------------------------------------+
+
+    Args:
+        interval (str, optional): Time string specifying how often to log train images. For example, ``interval='1ep'``
+            means images are logged once every epoch, while ``interval='100ba'`` means images are logged once every 100
+            batches. Eval images are logged once at the start of each eval. Default: ``"100ba"``.
+        mode (str, optional): How to log the image labels. Valid values are ``"input"`` (input only)
+            and "segmentation" which also saves segmentation masks. Default: ``"input"``.
+        num_images (int, optional): Number of images to log. Should be less than or equal to than the microbatch size.
+            If there are not enough images in the microbatch, all the images in the microbatch will be logged.
+            Default: ``8``.
+        input_key (str | int | Tuple[Callable, Callable] | Any, optional): A key that indexes to the input
+            from the batch. Can also be a pair of get and set functions, where the getter
+            is assumed to be first in the pair.  The default is 0, which corresponds to any sequence, where the first
+            element is the input. Default: ``0``.
+        target_key (str | int | Tuple[Callable, Callable] | Any, optional): A key that indexes to the target
+            from the batch. Can also be a pair of get and set functions, where the getter
+            is assumed to be first in the pair. The default is 1, which corresponds to any sequence, where the second
+            element is the target. Default: ``1``.
+    """
+
+    def __init__(self,
+                 interval: str = '100ba',
+                 mode: str = 'input',
+                 num_images: int = 8,
+                 input_key: Union[str, int, Tuple[Callable, Callable], Any] = 0,
+                 target_key: Union[str, int, Tuple[Callable, Callable], Any] = 1):
+        self.interval = interval
+        self.mode = mode
+        self.num_images = num_images
+        self.input_key = input_key
+        self.target_key = target_key
+
+        try:
+            import wandb
+        except ImportError as e:
+            raise MissingConditionalImportError(extra_deps_group='wandb',
+                                                conda_package='wandb',
+                                                conda_channel='conda-forge') from e
+        del wandb  # unused
+
+        # Check that the output mode is valid
+        if self.mode.lower() not in ['input', 'segmentation']:
+            raise ValueError(f'Invalid mode: {mode}')
+
+        # Check that the interval timestring is parsable and convert into time object
+        try:
+            self.interval = Time.from_timestring(interval)
+        except ValueError as error:
+            raise ValueError(f'Invalid time string for parameter interval') from error
```

Comment:
nit: no need to re-raise the exception; the stack trace should make it obvious that it was the parameter interval that was a bad timestring

Reply:
Cleaned this up